src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -12,3 +23,799 @@ pub mod crio;
+        Ok(serde_json::from_reader(f)?)
+    }
+
+    /// Get an immutable reference to the annotation hashmap.
+    pub fn get_annotation(&self) -> &HashMap<String, String> {
+        &self.annotations
+    }
+
+    /// Get a mutable reference to the annotation hashmap.
+    pub fn get_annotation_mut(&mut self) -> &mut HashMap<String, String> {
+        &mut self.annotations
+    }
+
+    /// Get the value of annotation with `key` as string.
+    pub fn get(&self, key: &str) -> Option<String> {
+        let v = self.annotations.get(key)?;
+        let value = v.trim();
+
+        if !value.is_empty() {
+            Some(String::from(value))
+        } else {
+            None
+        }
+    }
+
+    /// Get the value of annotation with `key` as bool.
+    pub fn get_bool(&self, key: &str) -> Option<bool> {
+        if let Some(value) = self.get(key) {
+            let value = value.trim();
+            if value == "true" {
+                return Some(true);
+            } else if value == "false" {
+                return Some(false);
+            } else {
+                warn!(sl!(), "failed to parse bool value from {}", value);
+            }
+        }
+
+        None
+    }
+
+    /// Get the value of annotation with `key` as u32.
+    pub fn get_u32(&self, key: &str) -> Option<u32> {
+        let s = self.get(key)?;
+        match s.parse::<u32>() {
+            Ok(nums) => {
+                if nums > 0 {
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u32 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+
+    /// Get the value of annotation with `key` as i32.
+    pub fn get_i32(&self, key: &str) -> Option<i32> {
+        let s = self.get(key)?;
+        match s.parse::<i32>() {
+            Ok(nums) => {
+                if nums > 0 {
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u32 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+
+    /// Get the value of annotation with `key` as u64.
+    pub fn get_u64(&self, key: &str) -> Option<u64> {
+        let s = self.get(key)?;
+        match s.parse::<u64>() {
+            Ok(nums) => {
+                if nums > 0 {
+                    Some(nums)
+                } else {
+                    None
+                }
+            }
+
+            Err(e) => {
+                warn!(
+                    sl!(),
+                    "failed to parse u64 value from {}, error: {:?}", s, e
+                );
+                None
+            }
+        }
+    }
+    /*
```

Comment:
Delete or add a TODO for functions that need to be implemented.

Reply:
delete it already